### ISSUE TITLE
Howto disappeared from menu

### ISSUE THREAD

**Issue by JanWielemaker** [Anniepoo/swiplsitedesign#32]
Some of the material in /howto/ is now linked from TUTORIALS, but not all.  I think
that the tutorial ones should cover only the highlights?  That would mean we still
need a HOWTO.

I'm tempted to add a HOWTO below FAQ in DOCUMENTATION, but Aniepoo
put a lot of thought in the main menu and I don't want to turn it into a mess
again.  Anniepoo?

P.s.  I think a HOWTO article is the way to solve
http://www.swi-prolog.org/bugzilla/show_bug.cgi?id=156

**Comment by Anniepoo** [comment]
The material in the howto's is supposed to be distributed into the new tutorials pages. I've not gotten there yet.
Lets live with the howto's being hard to find for a couple weeks while this all gets sorted. 
I'm waiting for Randy to get back on 27th, then he and I will focus on getting all those static information pages 
spiffied up. 

The fix for bug 156 is to 
a) Improve the pldoc comment in file_search_path 
b) close bug 3 in swiplwebtut  https://github.com/Anniepoo/swiplwebtut/issues/3
and move the web tutorial into swi-prolog.org  (planned change)

**Comment by JanWielemaker** [comment]
Allright.  Dealing with the file search mechanism is complicated stuff that is more suitable for
a HOWTO though (in addition to re-arranging and fixing the docs).  Similar issues are the
use of the print_message/2 infrastructure.  There are already a couple of old HOWTOS
that I'd like to be accessible and which currently are not, such as `how to submit patches'.

Isn't the menu going to be too big?

**Comment by Anniepoo** [comment]
Yes, a separate tutorial would be good. The tutorials menu's divided into basic/intermediate/advanced just because of length issues.  If we outgrow what fits on the menu we'll have to find a solution, but for now we're OK on length, I think. I want to avoid going back to a 'bouncer' page if we can.
A good problem to have would be to have so many of these conceptual overview/howto sections that they need to be incorporated in the search.

Jan, can you get me a list of which howtos aren't accessible that you want to be? 
The menu structure's not 'holy', by any means. If we need to change it we change it. The key is avoiding sliding back into sticking things in corners. It's like the difference between a messy house where no one dare clean up my desk or I'll never find anything and the house where it always seems they're about to shoot a commercial in there. We want to be the latter.

The 'how to submit a patch' could be linked off git repository page, which will have instructions when I _finally_ finish it.

**Comment by JanWielemaker** [comment]
I think that basically anything that is in /howto/ needs a place.  Making them appear in search would be a
definite gain.  I don't really agree with the `bouncer page'.  Overview pages are pretty useful as you can
search them and notably nested menus are hard to navigate.  Especially when using a toplevel stating
the`level'.  That might not be the right entry point.

Also, /howto/ is a wiki page, which allows people to add new howtos.  It would be great if that really
happened ...  Putting it in the menu however makes this practically impossible.

**Comment by Anniepoo** [comment]
I see your point. What you're calling an overview page I'm calling a 'bouncer page' - a page with no content of it's own, but a bunch of links.
My concern is that when we started the website was a maze of these things. It might have made some logical or programmatic sense, but in practice it was hard for people to use.

In practice there's some Pareto distribution of frequency of use of tutorials. Beginner tutorials and the web tutorial get used frequently, 'how to submit a patch' less so. Maybe a solution is best of both worlds - go ahead and have a 'tutorials' or 'howto' page, off a 'More Howtos' menu, and put the most common ones on the menu.

I don't like this solution because it requires thinking on the user's part, which violate's Krug's rule #1, don't make the user think, and because it hides stuff under 'more here'....

I don't have a great solution for this. I'm going to ask on stackexchange's UX board and see what they say.
Give me a bit, I'll work this problem.

**Comment by Anniepoo** [comment]
I asked about this on stackexchange.

http://ux.stackexchange.com/questions/50859/

**Comment by JanWielemaker** [comment]
For now, added the links for submitting patches and add-ons to COMMUNITY.  It seems right to me to have bug reporting and patch submission next to each other.  Submitting add-ons (packs) is a clear community task.  This howto is also linked from the add-on page, but that link should be made more clear.

This doesn't solve the general issue of adding howtos/tutorials.   In theory, we could use the navigation tree for that, which would be close to the option suggested in the link above.  The practical problem though is that some
of these tutorials are manual pages, some are wikis and yet others are external links.   We could get rid of the the manual pages by rewriting these as wikis.  Then we could have a general wiki navigation tree that uses the
directory hierarchy.  That might solve most of this, except for external tutorials.  Possibly we could hack these
using an iframe?  Messy stuff :-(

**Comment by Anniepoo** [comment]
Yes, and you've now got community over the 8 element limit for menu length, Your fix at least gets the links somewhere. Soon as I get out of the download hellhole this is next project (well, little matter of making a living may
get in the way).

**Comment by JanWielemaker** [comment]
Good that I didn't study HCI :-)  I'm not too dogmatic about 8 or 9.  I agree we're reaching the limit though.
Good news is that the new search makes life a lot easier.

**Comment by Anniepoo** [comment]
it's not a hard rule. Jessica & I ended up violating it ourselves, choosing 'put where belongs' over length.

But, for example, this is why contributors and publications ended up in documentation instead of community.

Silly question, isn't there some HCI group at VNU?

**Comment by JanWielemaker** [comment]
Hmm.  What about deleting Support and Contact from COMMUNITY and adding Contributors?  That would
not only get us back to 8, but also put the Contributors where they belong.  Publications is fine, as this is a
form of documentation.

**Comment by Anniepoo** [comment]
LOL - I've created an info design monster! Jan's moving menus around!
Good idea, but where's support and contact go?

**Comment by JanWielemaker** [comment]
Support is basically one of your beloved `bouncer pages', where everything it points to
is already in some menu.  To some extend that is also true for Contact, although there
is some info there and anyone expects having a contact menu entry.  Maybe it could
move somewhere else though; the contact page is specifically not intended for 
community.   But, ... where?

**Comment by Anniepoo** [comment]
I'm not opposed to repeating info when that's the clearest thing to do.
The support page purportedly serves this function - I'm Sally Noob, I can't figure out X random question, where do I go for help? I distinctly remember using that page as a noobie.  BUT, the page is now effectively the 'community menu. So yup, I'm OK with killing it.

Suggestion - you might consider simply killing the contact line. Do you actually have lots of people who have such a relationship with SWI-Prolog that they need to contact you directly, yet don't HAVE your email address, nor know how to get it? I assume info@swi-prolog.org forwards to you.

How about putting your email on the contributors page and dropping the contact page? After all, you're in a sense a contributor. And, as we move forward, eventually it'll be more appropriate to contact other developers when we're closer to having a true dev team, not just Jan and the seven dwarves.
